gax-grpc/src/main/java/com/google/api/gax/grpc/ProtoOperationTransformers.java

```diff
@@ -55,24 +55,13 @@ private ResponseTransformer(Class<ResponseT> packedClass) {
     public ResponseT apply(OperationSnapshot operationSnapshot) {
       if (!operationSnapshot.getErrorCode().getCode().equals(Code.OK)) {
         throw ApiExceptionFactory.createException(
-            "Operation with name \""
-                + operationSnapshot.getName()
-                + "\" failed with status = "
-                + operationSnapshot.getErrorCode(),
-            null,
-            operationSnapshot.getErrorCode(),
-            false);
+            operationSnapshot.getErrorMessage(), null, operationSnapshot.getErrorCode(), false);
       }
       try {
         return transformer.apply((Any) operationSnapshot.getResponse());
       } catch (RuntimeException e) {
         throw ApiExceptionFactory.createException(
-            "Operation with name \""
-                + operationSnapshot.getName()
-                + "\" succeeded, but encountered a problem unpacking it.",
-            e,
-            operationSnapshot.getErrorCode(),
-            false);
```

Comment:
This code path is explicitly when the operation code is `OK`, meaning there is no operation error.